src/storm-pomdp/analysis/IterativePolicySearch.cpp

```diff
@@ -238,34 +234,36 @@ namespace storm {
                 for (uint64_t action = 0; action < pomdp.getNumberOfChoices(state); ++action) {
                     std::vector<storm::expressions::Expression> subexprreachSwitch;
                     std::vector<storm::expressions::Expression> subexprreachNoSwitch;
+
                     subexprreachSwitch.push_back(!reachVarExpressions[state]);
                     subexprreachSwitch.push_back(!actionSelectionVarExpressions[pomdp.getObservation(state)][action]);
                     subexprreachSwitch.push_back(!switchVarExpressions[pomdp.getObservation(state)]);
+                    subexprreachSwitch.push_back(followVarExpressions[pomdp.getObservation(state)]);
+
                     subexprreachNoSwitch.push_back(!reachVarExpressions[state]);
                     subexprreachNoSwitch.push_back(!actionSelectionVarExpressions[pomdp.getObservation(state)][action]);
                     subexprreachNoSwitch.push_back(switchVarExpressions[pomdp.getObservation(state)]);
-                    subexprreachSwitch.push_back(followVarExpressions[pomdp.getObservation(state)]);
                     subexprreachNoSwitch.push_back(followVarExpressions[pomdp.getObservation(state)]);
+
                     for (auto const &entries : pomdp.getTransitionMatrix().getRow(rowindex)) {
-                        if (pomdp.getObservation(entries.getColumn()) == pomdp.getObservation(state)) {
+                        if (!delayedSwitching || pomdp.getObservation(entries.getColumn()) != pomdp.getObservation(state)) {
                             subexprreachSwitch.push_back(continuationVarExpressions.at(entries.getColumn()));
                         } else {
                             subexprreachSwitch.push_back(reachVarExpressions.at(entries.getColumn()));
+                            subexprreachSwitch.push_back(continuationVarExpressions.at(entries.getColumn()));
```

Comment:
Is it intended that this line is the same as in the if-branch?

Reply:
Yes, added a comment.